spanner/transaction.go

```diff
@@ -223,27 +236,46 @@ func (t *txReadOnly) ReadRowUsingIndex(ctx context.Context, table string, index
 	}
 }
 
-// QueryOptions provides options for executing a sql query from a database.
+// QueryOptions provides options for executing a sql query or update statement.
 type QueryOptions struct {
-	Mode    *sppb.ExecuteSqlRequest_QueryMode
-	Options *sppb.ExecuteSqlRequest_QueryOptions
+	Mode       *sppb.ExecuteSqlRequest_QueryMode
+	Options    *sppb.ExecuteSqlRequest_QueryOptions
+	RequestTag string
+	// transactionTag is unexported as applications should set this on the transaction and
+	// the transaction will then set this field for each request during the transaction.
+	transactionTag string
```

Comment:
Why don't we put this to `TransactionOptions`? `transactionTag` seems to be more tight with a transaction instead of a query/update?

Reply:
That's a reasonable question. I'll look into that to see whether that would be more reasonable once the proto changes have been merged. I tried to re-generate locally, but that seems to fail at the moment.

Reply:
I've removed the `transactionTag` from the `QueryOptions` as it was not a logical place to keep it.